eessi/testsuite/tests/apps/espresso/espresso.py

```diff
@@ -32,7 +32,7 @@ def filter_scales_P3M():
 
 
 @rfm.simple_test
-class EESSI_ESPRESSO_P3M_IONIC_CRYSTALS(rfm.RunOnlyRegressionTest):
+class EESSI_ESPRESSO(rfm.RunOnlyRegressionTest):
```

Comment:
I'm not sure if I would combine these in a single class, both tests don't have _that_ much in common. The part that they _do_ have in common, could be put in a parent class `EESSI_ESPRESSO` from which you could then make `EESSI_ESPRESSO_P3M_IONIC_CRYSTALS` inherit.

I know that in the GROMACS test all is put in a single class, but I think there is more overlap there. Here, e.g. I'm not sure if `filter_scales_P3M` would also apply to `lj`, it might scale better than the P3M case and thus allow us to include a 16-node run. Also, time limit that we'd need is different (I guess). `default_strong_scaling_system_size` only applies to `P3M`, (right?), etc.

Reply:
Note: it'll save us a lot of if-else's if we separate it in two classes. Also, it's more scalable: if we have 10 cases later on, the if-ifelse-ifelse will get confusing :)